pagopa/io-app
# feat: [IOCOM-164] Replaces custom types and decoders on io-backend client

## Short description
This PR replaces some custom types and decoders on four backend defined services.

## List of changes proposed in this pull request
- `ts/api/backend.ts`: replaced custom types with autogenerated ones on `getMessagesT` and `getMessageT`. Replaced custom decoder with autogenerate one on `postStartEmailValidationProcessT` and `abortUserDataProcessingT`

## How to test
- Using the dev server, try to get the message list and some standard message details. For both calls, test the following http response codes (using `messages.response.getMessagesResponseCode` and `messages.response.getMessageResponseCode` in the dev-server `config` file): 200, 400, 401, 404, 429, 500. Each one should be properly handled in app.
- Do the same for the email validation (edit `is_email_validated` in `src/payloads/profile.ts` on the dev-server and do a login), status codes 202, 400, 404, 429 and 500.
- Do the same for the cancel user profile deletion (go to the profile in app, request data deletion. After logout, login again, go to the profile and request a cancelling of user profile deletion), status codes 202, 400, 401, 404, 409, 429, 500.
